src/Akka.Persistence.Sql/Query/Dao/BaseByteReadArrayJournalDao.cs

```diff
@@ -82,54 +84,56 @@ public abstract class BaseByteReadArrayJournalDao : BaseJournalDaoWithReadMessag
                                 {
                                     return await connection
                                         .GetTable<JournalRow>()
                                         .Where(
                                             r =>
-                                                r.Tags.Contains(tagValue) &&
+                                                r.Tags.Contains(inVals.Tag) &&
                                                 !r.Deleted &&
-                                                r.Ordering > input.offset &&
-                                                r.Ordering <= input.maxOffset)
+                                                r.Ordering > inVals.Offset &&
+                                                r.Ordering <= inVals.MaxOffset)
                                         .OrderBy(r => r.Ordering)
-                                        .Take(input.maxTake)
+                                        .Take(inVals.Max)
                                         .ToListAsync(token);
                                 });
                         })
                     .Via(_deserializeFlow),
 
                 TagMode.TagTable => AsyncSource<JournalRow>
                     .FromEnumerable(
-                        new { separator, tag, offset, maxOffset, maxTake, _connectionFactory = ConnectionFactory },
-                        async input =>
+                        new { inst=this, args= new QueryArgs(offset,maxOffset,maxTake,tag,TagMode.TagTable)},
+                        static async input =>
                         {
-                            return await input._connectionFactory.ExecuteWithTransactionAsync(
-                                ReadIsolationLevel,
-                                ShutdownToken,
-                                async (connection, token) =>
+                            var inst = input.inst;
+                            return await inst.ConnectionFactory.ExecuteWithTransactionAsync(
+                                input.args,
+                                inst.ReadIsolationLevel,
+                                inst.ShutdownToken,
+                                static async (connection, token,txInput) =>
                                 {
-                                    var journalTable = connection.GetTable<JournalRow>();
-                                    var tagTable = connection.GetTable<JournalTagRow>();
-
                                     var query =
-                                        from r in journalTable
-                                        from lp in tagTable.Where(jtr => jtr.OrderingId == r.Ordering).DefaultIfEmpty()
-                                        where lp.OrderingId > input.offset &&
-                                              lp.OrderingId <= input.maxOffset &&
+                                        from r in connection.GetTable<JournalRow>()
```

Comment:
It would still be nice to convert this to method syntax later, if only because it's something like 300 lines of IL as it stands 😅